datagen/gtfs/service_levels.py

```diff
@@ -41,6 +42,20 @@ def bucket_trips_by_hour(trips: List[TripSummary]):
         by_time_of_day[hour] += 0.5
     return by_time_of_day
 
+# Given a list of trips on a single date, prune to match a more realistic service level
+def prune_trips_single_date(trips: List[TripSummary]):
```

Comment:
How about `filter_trips_serving_most_common_stop_on_route()`? Also, the comments are good (better than any I've added to this whole repo) but I think they could allude more directly to the problem this function solves.